### PR TITLE
Add Boot build and fix specs

### PR DESCRIPTION
**Pull request by frenchy64**
This PR fixes the specs for internal functions so that the unit tests can run successfully with `instrument`. This is tested with `in-n-outstrument`, and the following specs are not called by the unit+integration tests:

```
                             - sails-forth.client/count!
                             - sails-forth.client/create!
                             - sails-forth.client/describe!
                             - sails-forth.client/get!
                             - sails-forth.client/get-attrs-for-label
                             - sails-forth.client/get-fields
                             - sails-forth.client/import!
                             - sails-forth.client/limits!
                             - sails-forth.client/list!
                             - sails-forth.client/objects!
                             - sails-forth.client/put!
                             - sails-forth.client/query!
                             - sails-forth.client/schema
                             - sails-forth.http/add-import-batch!
                             - sails-forth.http/close-import-job!
                             - sails-forth.http/count!
                             - sails-forth.http/create!
                             - sails-forth.http/create-import-job!
                             - sails-forth.http/delete!
                             - sails-forth.http/limits!
                             - sails-forth.http/list!
                             - sails-forth.http/update!
                             - sails-forth.query/record-type-id
                             - sails-forth.query/record-types
                             - sails-forth.update/create!
                             - sails-forth.update/import!
                             - sails-forth.update/update!
```

### CODE REVIEW

File: build.boot [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ '[[adzerk/boot-jar2bin "1.1.0" :scope "build"]
+   [adzerk/boot-test "1.2.0" :scope "test"]
+   [big-solutions/boot-mvn "0.1.5"]
+   [sparkfund/spec-coverage "0.1.0" :scope "test"]
+   [org.clojure/test.check "0.9.0" :scope "test"]
+   [cheshire "5.5.0"]
+   [clj-http "2.0.0"]
+   [clj-time "0.11.0"]
+   [com.github.jsqlparser/jsqlparser "0.9.5"]
+   [org.clojure/clojure "1.9.0-alpha10"]])
```
Comment (dball) [inline build.boot]: An annoying thing I'm super pedantic about is sorting lists like this in... a consistent order; alphabetically is the most obvious. :P

File: src/sails_forth/http.clj [hunk 2]
```diff
@@ -94,7 +94,7 @@
 (s/def ::host
   string?)
 
-(s/def read-only?
+(s/def ::read-only?
```
Comment (dball) [inline src/sails_forth/http.clj:100]: :joy:

File: src/sails_forth/spec.clj [hunk 3]
```diff
@@ -48,14 +50,96 @@
...
+                   ::picklistValues
+                   ::name
+                   ::referenceTo
+                   ::scale
+                   ::precision
+                   ::label
+                   ::relationshipName]))
+
+(s/def ::field-description
+  (s/multi-spec field-description-type :type))
```
Comment (dball) [inline src/sails_forth/spec.clj:184]: Very nice. That's actually pretty good as documentation.

File: test/sails_forth/clojurify_test.clj [hunk 4]
```diff
@@ -36,6 +37,7 @@
                                 6.18M))))
   (testing "int"
     (is (= 500 (parse-value {:type "int"} 500M))))
+  #_
```
Comment (dball) [inline test/sails_forth/clojurify_test.clj]: Um

File: test/sails_forth/clojurify_test.clj [hunk 5]
```diff
@@ -57,6 +59,7 @@
                           :precision 4}
                          0.0618M)
            6.18M)))
+  #_
```
Comment (dball) [inline test/sails_forth/clojurify_test.clj]: Er

**Review by dball — APPROVED** [review]
Super into this. I'd say go ahead and delete those commented tests since they're exercising the fn outside of its domain and I don't really want to specify its behavior thereby.

I'll give it a whirl locally, but nothing herein strikes me as hinky.

File: build.boot (hunk 1 above)
Comment (frenchy64) [reply build.boot]: Ok fixed.

**Comment by frenchy64** [comment]
Ok, I removed the bad tests.

File: src/sails_forth/http.clj [hunk 6]
```diff
@@ -66,10 +66,13 @@
 (s/def ::access_token
   string?)
 
-(s/def ::authentication
+(s/def ::non-nil-authentication
```
Comment (dball) [inline src/sails_forth/http.clj:69]: Eh, hate to bikeshed names, but specs maybe merit it... I guess I'd call this one ::authentication and I would use an anonymous spec `(s/nilable ::authentication) in the return value from `authentication`

File: src/sails_forth/spec.clj [hunk 7]
```diff
@@ -137,6 +143,42 @@
...
+                   ::relationshipName]))
+(defmethod field-description-type "phone" [_] 
+  (s/keys :req-un [::type
+                   ::picklistValues
+                   ::name
+                   ::referenceTo
+                   ::scale
+                   ::precision
+                   ::label
+                   ::relationshipName]))
```
Comment (dball) [inline src/sails_forth/spec.clj:181]: Good, well-chosen subset. We could probably go farther for some of these types and be prescriptive about the expected values also, but this is super fine for this iteration.

**Review by dball — APPROVED** [review]

File: src/sails_forth/http.clj (hunk 6 above)
Comment (frenchy64) [reply src/sails_forth/http.clj:69]: The reason for these acrobatics is that `::state` uses an `:authentication` entry, and in that context it's nilable. So this is the only combination that seems to work without using another namespace prefix.

File: src/sails_forth/spec.clj (hunk 7 above)
Comment (frenchy64) [reply src/sails_forth/spec.clj:181]: Thanks!

**Pull request merged**